#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/RectangleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.49993553657241463 0.0009091111687645141 -0.8660621414130366 3.1384372070321844
  position 48.82325337888147 -47.477340148425284 8.88723661565131
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 43.0631 -46.5896 4.78728e-13
  rotation 0 1 0 0
  floorSize 6 6
}
Robot {
  translation 43.5953215029365 -46.51184254026449 0.10018042027258131
  rotation 0.030943310850176077 -0.01700637075911566 0.9993764530282034 0.546248941078162
  children [
    DistanceSensor {
      translation 0.15 0 0.045
      rotation 0 1 0 1.5708
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_middle"
      boundingObject USE sensor
      physics DEF sensor_ph Physics {
      }
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
    DistanceSensor {
      translation 0.15 -0.07 0.045
      rotation 0 1 0 1.5708
      children [
        USE sensor
      ]
      name "ds_right"
      boundingObject USE sensor
      physics USE sensor_ph
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
    DistanceSensor {
      translation 0.15 0.07 0.045
      rotation 0 1 0 1.5708
      children [
        USE sensor
      ]
      name "ds_left"
      boundingObject USE sensor
      physics USE sensor_ph
      lookupTable [
        0 0 0
        0.2 1000 0
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 10.027824515660535
        axis 0 1 0
        anchor 0.07 0.11 -0.04
      }
      device [
        RotationalMotor {
          name "wheel_l1"
        }
      ]
      endPoint Solid {
        translation 0.070027267652764 0.11004491660482785 -0.03994964991182746
        rotation 0.21245392935820465 -0.6907665561997758 0.6911619873344895 2.722810960546835
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.06
            }
          }
        ]
        boundingObject USE wheel
        physics DEF sensor Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 10.027838400087
        axis 0 1 0
        anchor -0.07 0.11 -0.04
      }
      device [
        RotationalMotor {
          name "wheel_l2"
        }
      ]
      endPoint Solid {
        translation -0.07000174368178091 0.11000051384052512 -0.04000113711457293
        rotation 0.21245872273956254 -0.6907649359804362 0.6911621331871542 2.7228009102107174
        children [
          USE wheel
        ]
        boundingObject USE wheel
        physics USE sensor
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 8.661360631626824
        axis 0 1 0
        anchor -0.07 -0.11 -0.04
      }
      device [
        RotationalMotor {
          name "wheel_r2"
        }
      ]
      endPoint Solid {
        translation -0.07003896876627148 -0.11004390836841083 -0.03995887961910972
        rotation 0.275028729835327 0.6797095328575882 -0.6799662849786642 2.6046095217284644
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics USE sensor
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 8.661342919979251
        axis 0 1 0
        anchor 0.07 -0.11 -0.04
      }
      device [
        RotationalMotor {
          name "wheel_r1"
        }
      ]
      endPoint Solid {
        translation 0.07000168344890494 -0.11000029831917857 -0.04000096932494218
        rotation 0.27503567111947214 0.6797089250803502 -0.6799640849177085 2.6045984818830195
        children [
          USE wheel
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics USE sensor
      }
    }
    DEF robot_body Group {
      children [
        Transform {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.3 0.8 0.4
                metalness 0
              }
              geometry Box {
                size 0.25 0.2 0.1
              }
            }
          ]
        }
        Transform {
          translation 0.01 0 0.05
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.3 0.8 0.4
                metalness 0
              }
              geometry Box {
                size 0.3 0.3 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  boundingObject USE robot_body
  physics Physics {
  }
  controller "linefollower_controller"
}
Solid {
  translation 43.6601 -46.8911 3.55272e-15
  rotation 0 0 1 4.692820414042842e-06
  scale 0.035 0.035 0.035
  children [
    DEF line Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
      }
      geometry Mesh {
        url [
          "../libraries/line.obj"
        ]
      }
    }
  ]
  name "track"
  boundingObject USE line
}
